activesupport/lib/active_support/cache.rb

```diff
@@ -311,29 +312,31 @@ def mute
       #     :bar
       #   end
       #   cache.fetch('foo') # => "bar"
-      def fetch(name, options = nil)
+      def fetch(raw_key, options = nil)
+        master_key = KeyMaster.to_key(raw_key)
```

Comment:
In most methods you can safely replace the incoming name variable with an instance of KeyMaster, however here we are using the raw key by passing it into the block so it cannot change. 

After that I wanted to preserve passing in a KeyMaster object to all places that would otherwise have taken a `name`.

Naming is hard. I'm open to suggestions.